client/editor/editor.go

```diff
@@ -126,18 +126,142 @@ func (e *Editor) showPositions() {
 }
 
 // MoveCursor updates the Cursor position.
-func (e *Editor) MoveCursor(x, _ int) {
+func (e *Editor) MoveCursor(x, y int) {
+	// move cursor horizontally
 	newCursor := e.Cursor + x
 
-	if newCursor < 0 {
-		newCursor = 0
+	// move cursor vertically
+	if y > 0 {
+		newCursor = e.calcCursorDown()
+	}
+
+	if y < 0 {
+		newCursor = e.calcCursorUp()
 	}
+
```

Comment:
Could this be condensed into a if-else block?

Reply:
I thought keeping them separate might be preferable because we want to do nothing if y == 0.

Reply:
Ah, makes complete sense. Let's keep it as it is.